gluoncv/data/tracking_data/track.py

```diff
@@ -0,0 +1,779 @@
+        self.pick = self.shuffle()
+
+    def shuffle(self):
+        pick = []
+        m = 0
+        while m < self.num:
+            p = []
+            for sub_dataset in self.all_dataset:
+                sub_p = sub_dataset.pick
+                p += sub_p
+            np.random.shuffle(p)
+            pick += p
+            m = len(pick)
+        self.logger.info("shuffle done!")
+        self.logger.info("dataset length %d", self.num)
+        return pick[:self.num]
+
+    def _find_dataset(self, index):
+        for per_dataset in self.all_dataset:
+            if per_dataset.start_idx + per_dataset.num > index:
+                return per_dataset, index - per_dataset.start_idx
+        return None, None
+
+    def _get_bbox(self, image, shape):
+        imh, imw = image.shape[:2]
+        if len(shape) == 4:
+            w, h = shape[2]-shape[0], shape[3]-shape[1]
+        else:
+            w, h = shape
+        context_amount = 0.5
+        exemplar_size = self.train_exemplar_size
+        wc_z = w + context_amount * (w+h)
+        hc_z = h + context_amount * (w+h)
+        s_z = np.sqrt(wc_z * hc_z)
+        scale_z = exemplar_size / s_z
+        w = w*scale_z
+        h = h*scale_z
+        cx, cy = imw//2, imh//2
+        bbox = center2corner(Center(cx, cy, w, h))
+        return bbox
+
+    def __len__(self):
+        return self.num
+
+    def __getitem__(self, index):
+        index = self.pick[index]
+        data, index = self._find_dataset(index)
+
+        gray = self.gray and self.gray > np.random.random()
+        neg = self.neg and self.neg > np.random.random()
+
+        # get one dataset
+        if neg:
+            template = data.get_random_target(index)
+            search = np.random.choice(self.all_dataset).get_random_target()
+        else:
+            template, search = data.get_positive_pair(index)
+
+        # get image
+        template_image = self.cv2.imread(template[0])
+        search_image = self.cv2.imread(search[0])
+
+        # get bounding box
+        template_box = self._get_bbox(template_image, template[1])
+        search_box = self._get_bbox(search_image, search[1])
+
+        # augmentation
+        template, _ = self.template_aug(template_image,
+                                        template_box,
+                                        self.train_exemplar_size,
+                                        gray=gray)
+
+        search, bbox = self.search_aug(search_image,
+                                       search_box,
+                                       self.train_search_size,
+                                       gray=gray)
+
+        # get labels
+        cls, delta, delta_weight, _ = self.anchor_target(bbox, self.train_output_size, neg)
+        template = template.transpose((2, 0, 1)).astype(np.float32)
+        search = search.transpose((2, 0, 1)).astype(np.float32)
+
+        return template, search, cls, delta, delta_weight, np.array(bbox)
+
+class Augmentation:
```

Comment:
For augmentation function (image transformations), we usually put it inside folder `gluoncv.data.transforms`. You can start a new file, named `track.py`.